packages/react/src/components/Tag/Tag-test.js

```diff
@@ -10,10 +10,39 @@ import Tag from '../Tag';
 import TagSkeleton from '../Tag/Tag.Skeleton';
 import { shallow } from 'enzyme';
 import { settings } from 'carbon-components';
+import { render, cleanup } from '@carbon/test-utils/react';
 
 const { prefix } = settings;
 
 describe('Tag', () => {
+  afterEach(cleanup);
+
+  describe('automated accessibility testing', () => {
+    it('should have no Axe violations', async () => {
+      const { container } = render(<Tag>This is not a tag</Tag>);
+      await expect(container).toHaveNoAxeViolations();
+    });
+
+    it('should have no DAP violations', async () => {
+      const { container } = render(<Tag>This is not a tag</Tag>);
+      await expect(container).toHaveNoDAPViolations('Tag');
+    });
+  });
+
+  describe('with a screenreader', () => {
+    it('filtered variant should have appropriate aria-label', () => {
+      const { container } = render(<Tag filter>This is not a tag</Tag>);
+
+      const button = container.querySelector('[aria-label]');
+      expect(button).toBeInstanceOf(HTMLElement);
+
+      const ariaLabel = button.getAttribute('aria-label');
+      expect(ariaLabel).toBeDefined();
+
+      expect(ariaLabel).toEqual('Clear filter This is not a tag');
```

Comment:
I'm a little torn on this assertion just because `Clear filter` could change to whatever language we want, would it make sense for us to have the check be that the accessibility label just includes `children` text? Something like:

```jsx
const children = 'tag content';
const { container } = render(<Tag filter>{children}</Tag>);

// No need to support both, but might be worthwhile so we could use either `aria-label` or `aria-labelledby`
// here. Optionally, could add a DOM helper like `getAccessibleLabel` or `getLabel` or something else that works
const button = container.querySelector('[aria-label], [aria-labelledby]');

const accessibilityLabel = button.getAttribute('aria-label') || button.getAttribute('aria-labelledby');

// This check would mirror our "Accessibility label must contain at least all of visible label"
// requirement
expect(accessibilityLabel).toEqual(
  expect.stringContaining(children)
);
```

Reply:
Totally agree. And great thinking ahead 💯